Add support for XML using Nokogiri::XML parser

DanLindeman opened the pull request:
- Add tests for XML
- Add sample XML courtesy of Nokogiri demo at http://www.nokogiri.org/tutorials/searching_a_xml_html_document.html
- Add documentation to README.md

DanLindeman committed:
Add support for XML using Nokogiri::XML parser, Add tests for XML, add sample XML courtesy of Nokogiri demo at http://www.nokogiri.org/tutorials/searching_a_xml_html_document.html, Add documentation to README.md

DanLindeman committed:
add usage example to README.md

siadat commented:
Thank you for this PR! :+1:

siadat commented on README.md:
```diff
@@ -34,6 +34,7 @@ Data is parsed and stored as `x`:
     cat file.html | inspect html
     curl https://api.github.com/users/siadat/repos | inspect
 	curl https://api.github.com/repos/siadat/inspect/commits | inspect
+    inspect file.xml
```
I think `parse_data` function is still [detecting](https://github.com/DanLindeman/inspect/blob/xml_support/lib/inspect.rb#L8) this as HTML, instead of XML. Do you think there is any way to distinguish the two? I can only think of using [strict](http://www.nokogiri.org/tutorials/ensuring_well_formed_markup.html) mode to raise a SyntaxError when XML is not well formatted.

DanLindeman commented:
I noticed that, and I'll look into ways to fix that.

DanLindeman commented:
Turns out the xml case wasn't able to be called. I'll fiddle with the input parsing to allow xmls.

DanLindeman commented:
So, I came up with a solution for several issues I was having integrating XML support. 
- I added a function to infer the type passed to us from the filename (for the `inspect somefile.extension` use-case)
- I added a check to see if an xml doc was errantly parsed as html. I do this by checking if x.children.first.name == 'html' (which I believe implies errant conversion).

The take away is that we cannot support the `cat FILE | inspect [FORMAT]` use-case with this implementation. Let me know if you can think of a workaround.

DanLindeman committed:
Add format inference, allow correct html-xml decision, add noblanks config to xml, update example.xml, tests, gemspec, and README

DanLindeman committed:
Inital drop for yaml support, Add parse case, add example.yml, add tests, add data inferred loading, update gemspec and README

siadat commented:
Thank you for the update! :+1: A few suggestions:
- Use long-options for explicit formats, e.g. `--json`, `--xml`, etc. Use [docopt](https://github.com/docopt/docopt.rb), I think it is readable and very easy to use. For example:
  
  ```
  Usage:
    inspect [--json|--csv|--html|--xml|--yml] [<filename>]
  ```
- Replace `x.children.first.name` with `x.root.name` to make sure it works for HTML documents that start with a `<!DOCTYPE>` declaration. Add the doctype declaration to `test/example.html`.
- Rescue `Psych::SyntaxError` when parsing YML in the `parse()` method.

Other than that it looks good! :smile:

DanLindeman commented:
Sorry about the double-pull request. Thanks for the feedback, I'll fix those last couple of things up, and hopefully we can close this pull request.

siadat commented:
:+1:

DanLindeman committed:
Rescue Psych::SyntaxError, add docopt, ensure doctype is supported
